central/scannerdefinitions/handler/handler.go

```diff
@@ -530,21 +431,24 @@ func (h *httpHandler) openMostRecentV4OnlineFile(_ context.Context, t updaterTyp
 	return nil, fmt.Errorf("unknown Scanner V4 updater type: %s", t)
 }
 
-// openMostRecentV4OfflineFile gets desired offline file from compressed bundle: offlineScannerV4DefinitionBlobName
-func (h *httpHandler) openMostRecentV4OfflineFile(ctx context.Context, t updaterType, opts v4OpenOpts) (*vulDefFile, error) {
+// openOfflineDefinitions gets desired offline file from compressed bundle.
+func (h *httpHandler) openOfflineDefinitions(ctx context.Context, t updaterType, opts openOpts) (*vulDefFile, error) {
 	log.Debugf("Getting v4 offline data for updater: type %s: options: %#v", t, opts)
-	openedFile, err := h.openOfflineBlob(ctx, offlineScannerV4DefinitionBlobName)
-	if err == nil && openedFile == nil {
-		log.Warnf("Blob %s does not exist", offlineScannerV4DefinitionBlobName)
-		return nil, errors.New("No valid scanner V4 file in offline mode")
+	openedFile, err := h.openOfflineBlob(ctx, opts.offlineBlobName)
+	if err != nil {
+		return nil, fmt.Errorf("opening offline definitions: %s: %w",
+			opts.offlineBlobName, err)
+	}
+	if openedFile == nil {
+		log.Warnf("Blob %s does not exist", opts.offlineBlobName)
+		return nil, nil
 	}
-
 	var offlineFile *vulDefFile
 	defer utils.IgnoreError(openedFile.Close)
 	switch t {
-	case mappingUpdaterType:
+	case mappingUpdaterType, v2UpdaterType:
 		// search mapping file
-		fileName := filepath.Base(opts.mappingFile)
+		fileName := filepath.Base(opts.fileName)
```

Comment:
This looks like will always vail for the V2 use case, `fileName` will always be the empty string, has this been tested?

Reply:
(hopefully) fixed this in a recent commit